api/types/network/endpoint.go

```diff
@@ -108,7 +108,7 @@ func (ipamConfig *EndpointIPAMConfig) Validate() error {
 	}
 	for _, addr := range ipamConfig.LinkLocalIPs {
 		if net.ParseIP(addr) == nil {
-			errs = append(errs, fmt.Errorf("invalid link-local IP address %s", addr))
+			errs = append(errs, fmt.Errorf("invalid link-local IP address: %s", addr))
```

Comment:
I wanted to suggest: could we use the underlying `netip.Parse` instead, so that we can return an actual error with details "what is invalid"??

... only to realise that's new in Go 1.21 https://github.com/golang/go/blob/2c1e5b05fe39fc5e6c730dd60e82946b8e67c6ba/src/net/ip.go#L489-L502

Reply:
Good catch!

> ... only to realise that's new in Go 1.21

`netip.ParseAddr` is actually available since go 1.18, so we can perfectly use it. But looking at the git blame, it looks like `net` parsers will use `netip` ones starting with go 1.21.

As a consequence, we might start accepting IPv6 addresses with scope zone (eg. `fe80::1cc0:3e8c:119f:c2e1%ens18`), whereas it's invalid. I need to double-check but we probably need a proper error for that.

Reply:
Huh, that's what errors `netip.ParseAddr(...)` produces: `ParseAddr("foo"): unable to parse IP`. So, it'd look like this if we use it here:

```
* invalid IPv4 address foo: ParseAddr("foo"): unable to parse IP
* invalid IPv6 address bar: ParseAddr("bar"): unable to parse IP
* invalid link-local IP address baz: ParseAddr("baz"): unable to parse IP
* invalid link-local IP address foobar: ParseAddr("foobar"): unable to parse IP
```

I think we should rather stick with `net.ParseIP`.

Reply:
Ah, yes, that doesn't provide much more context indeed. I was hoping it would give the user more context what's wrong. Thanks for looking!